Gophkeeper

andreevym opened the pull request.

andreevym committed:
init commit

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
init commit

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
init commit

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
added client, and tests handler,documentation and migration

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
updated README.md, change the item number, 4->2. Sign In for receiving token

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
remove unnecessary comments to golang code

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
Create go.yml

andreevym committed:
fix go version for build

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
up version for libraries

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
fix goimports

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
update github workflow

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
make golang tests

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
Update go.yml

Go Release Binary

andreevym committed:
Update go.yml

andreevym committed:
Update go.yml

release

andreevym committed:
Update go.yml

release: types: published, edited

andreevym committed:
Create release.yml

added workflow for release

andreevym committed:
Update go.yml

removed release from go.yaml

andreevym committed:
Update release.yml

andreevym committed:
change server http run and shutdown

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
Merge remote-tracking branch 'origin/main'

andreevym committed:
small renaming

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
fix pointer, rename hasher to hashService

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
added documentation for this application

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
added documentation for this application

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
change description when count of args is wrong

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
updated output and successColor, errorColor

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
added code documentation, rewrite client code

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
fixed integration tests

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
added upload file

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
added upload file

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
Update go.yml

added linter

andreevym committed:
fixed example

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
Merge remote-tracking branch 'origin/main'

andreevym committed:
fix security "require github.com/opencontainers/runc v1.1.14"

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
Merge branch 'main' into develop

andreevym committed:
#1 migration file name for users and vault

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
#1 fixed ci

Signed-off-by: Yury Andreev <andreevym@gmail.com>

avhitrov reviewed (changes requested):
Базовый механизм взаимодействия клиента и сервера понятен и реализован на уровне MVP, хорошо документирован и принимается. Что осталось не реализованным:
1. (Дополнительные требования) клиент должен давать пользователю возможность получить информацию о версии и дате сборки бинарного файла клиента.
2. Описанные типы хранимой информации и интерфейс CLI предполагают возможность с помощью набора команд сформировать, отправить на сервер, получить и расшифровать в человекопонятной форме:
- пары логин/пароль;
- произвольные текстовые данные;
- произвольные бинарные данные;
- данные банковских карт;
плюс произвольную текстовую информацию.

Клиентская часть в текущей реализации не предоставляет такого интерфейса. То есть, не хватает проработки внутренней структуры передаваемых от клиента серверу vault-объектов, преобразования из, скажем, набора ключей в бинарь и обратного преобразования из бинаря в человекопонятный вывод.

3. Также за бортом осталась синхронизация данных и организация некоторого внутреннего хранилища клиента, но на реализации этого функционала я настаивать не буду.

andreevym committed:
added methods
- store-login-password
- get-login-password
- store-text
- get-text
- store-binary
- get-binary
- store-card
- get-card

Signed-off-by: Yury Andreev <andreevym@gmail.com>

andreevym committed:
added methods
- store-login-password
- get-login-password
- store-text
- get-text
- store-binary
- get-binary
- store-card
- get-card

Signed-off-by: Yury Andreev <andreevym@gmail.com>

avhitrov reviewed (approved):
Требуемый функционал реализован. Поздравляю с успешной сдачей дипломной работы!
